File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode slowPtr = head, fastPtr = head;

        while (fastPtr != null && fastPtr.next != null) {
            fastPtr = fastPtr.next.next;
            if (fastPtr == slowPtr) {
                return true;
            }
            slowPtr = slowPtr.next;

        }

        return false;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        def determineSZ(head: Optional[ListNode]) -> int:
            size = 0

            curr = head
            while curr != None:
                size += 1
                curr = curr.next

            return size

        sz = determineSZ(head)
        remove = sz - n

        dummy = ListNode(0, head)
        prev = dummy
        curr = head

        counter = 0
        while curr != None:
            if counter == remove:
                prev.next = curr.next
                break
            else:
                counter += 1
                curr = curr.next
                prev = prev.next

        return dummy.next